More details about the VCR testing recording framework (#411)
microsoft/qdk-python: azure-quantum/tests/README.md

#### Cause ####
The way VCR works is by working like a HTTP proxy.

If the request (coming from the SDK api calls) is found (by matching the full URI and HTTP headers) on the recorded file, it will playback the corresponding response. Otherwise, if not found, it will attempt to do a live cal to the web API and will try to record the results at the end. But if there is a recording file (cassete) already, it will give the error you are mentioning.

The environment variables to control whether to playback records or run live and re-create them is just a little logic to decide if we want to delete the recording files first before running the tests (which will force a call to the live APIs and re-record the calls).

But it does not handle the case when the recording file is there but a specific HTTP request/response can’t be found in the existing recording, which gives the error you are seeing.

This error could also be caused if the recorded files are manually updated and do not really match the requests that the SDK will actually request.

#### Potential solutions ####

> 1) One way to remove the error is to delete the existing recording file and let it do all the live calls, and create a new recording file that contains all the requests/responses that the tests need. Then, next time, you should be able to simple playback the recordings with no errors.

Reviewer: ```suggestion
1) One way to remove the error is to delete the existing recording file and let it do all the live calls and create a new recording file that contains all the requests/responses that the tests need. After that, you should be able to simple playback the recordings with no errors.
```

> 2) If the error still persist after trying (1), then probably there is something unique in the URL or HTTP headers of the HTTP request that changes at every time you run the tests. In this case, we need to either make that thing constant in the tests, or, if they are genuinely unique, we need to replace that unique value in the request recording pipeline such that, at least in the recording file, it will be unique.

Reviewer: ```suggestion
2) If the error still persist after trying (1), then probably there is something unique in the URL or HTTP headers of the HTTP request that changes every time you run the tests. In this case, we need to either make that thing constant in the tests, or if they are genuinely unique, we need to replace that unique value in the request recording pipeline such that, at least in the recording file, it will be unique.
```
